### ISSUE TITLE
StdDev returns NaN when input array x has <= one element

### ISSUE THREAD

**Issue by wzzhu** [gonum/gonum#1568]
### What are you trying to do?

Computingn StdDev for input array of length 1. 

### What did you do?

```
package gonum_test
import (
	"testing"

	"github.com/gonum/stat"
)

func TestStdev(t *testing.T) {
	if v := stat.StdDev([]float64{1.0}, nil); v != 0 {
		t.Errorf("Expecting 0, but got %v", v)
	}
}
```

### What did you expect to happen?

Get 0 instead of NaN

### What actually happened?

Got Nan from StdDev

Caused by this line

		variance = (ss - compensation*compensation/float64(len(x))) / float64(len(x)-1)

### What version of Go and Gonum are you using?

go 1.15.6

### Does this issue reproduce with the current master?
Yes

**Comment by kortschak** [comment]
This is expected behaviour, the variance of a single value is not defined. Here is the equivalent code in R.

```
> sd(c(1))
[1] NA
```

We don't have a language level `NA` value, but `NaN` is a usable equivalent.

**Comment by romanwerpachowski** [comment]
We're missing the biased estimators for variance and standard deviation in stat/. I ddded them in #1570

**Comment by romanwerpachowski** [comment]
New methods are merged in. @wzzhu, use `stat.PopStdDev` instead of `stat.StdDev` to get the result you wanted.

**Comment by kortschak** [comment]
Those will be available in v0.9 which should be released reasonably soon. Otherwise you will need to use a pseudo version.
